### ISSUE TITLE
Installing ipython: as part of  jupyter and independently

### ISSUE THREAD

**Issue by alanwilter** [pipxproject/pipx#656]
Perhaps it's more a question, but I found it confusing.

I first try to install `jupyter`:
```bash
pipx install jupyter
``` 
but it has dependencies so I finally did:
```bash
pipx install jupyter --include-deps
```
Great, all working. I also noticed that `ipython` was installed as well. Fine.
But I did:
```bash
pipx install ipython
⚠️  File exists at /Users/alan/.local/bin/iptest and points to /Users/alan/.local/pipx/venvs/jupyter/bin/iptest, not /Users/alan/.local/pipx/venvs/ipython/bin/iptest. Not
    modifying.
⚠️  File exists at /Users/alan/.local/bin/iptest3 and points to /Users/alan/.local/pipx/venvs/jupyter/bin/iptest3, not /Users/alan/.local/pipx/venvs/ipython/bin/iptest3. Not
    modifying.
⚠️  File exists at /Users/alan/.local/bin/ipython and points to /Users/alan/.local/pipx/venvs/jupyter/bin/ipython, not /Users/alan/.local/pipx/venvs/ipython/bin/ipython. Not
    modifying.
⚠️  File exists at /Users/alan/.local/bin/ipython3 and points to /Users/alan/.local/pipx/venvs/jupyter/bin/ipython3, not /Users/alan/.local/pipx/venvs/ipython/bin/ipython3. Not
    modifying.
  installed package ipython 7.21.0, Python 3.9.2
    - iptest (symlink missing or pointing to unexpected location)
    - iptest3 (symlink missing or pointing to unexpected location)
    - ipython (symlink missing or pointing to unexpected location)
    - ipython3 (symlink missing or pointing to unexpected location)
done! ✨ 🌟 ✨
```
Also doing `pipx list` shows:
```
   package ipython 7.21.0, Python 3.9.2
    - iptest (symlink missing or pointing to unexpected location)
    - iptest3 (symlink missing or pointing to unexpected location)
    - ipython (symlink missing or pointing to unexpected location)
    - ipython3 (symlink missing or pointing to unexpected location)
   package jupyter 1.0.0, Python 3.9.2
    - iptest
    - iptest3
    - ipython
    - ipython3
    - jsonschema
    - jupyter
    - jupyter-bundlerextension
    - jupyter-console
    - jupyter-kernel
    - jupyter-kernelspec
    - jupyter-migrate
    - jupyter-nbconvert
    - jupyter-nbextension
    - jupyter-notebook
    - jupyter-qtconsole
    - jupyter-run
    - jupyter-serverextension
    - jupyter-troubleshoot
    - jupyter-trust
    - pygmentize
```
Nevertheless, everything is working, even after I've uninstalled `ipython` (`pipx uninstall ipython`). However, how to read these warnings?
Honestly, I would expect `pipx` **NOT** to install `ipython` by default if it detects as already installed, even if as a dependency.
If the user still want to install it, then another argument should be provided, like `--force`.

This not only about `jupyter` x `ipython`, other packages may be in this same situation.

BTW, this tool `pipx` is absolutely **_fantastic_**!

**Comment by itsayellow** [comment]
The short answer: if you do `pipx install jupyter --include-dependencies` then you don't need to separately install ipython to have ipython available in your PATH.

The longer answer: pipx always installs a package and all of its dependencies into its own directory, but what binaries are made available to a user in their PATH depends on a few factors.
* Normal `pipx install`: only the main package's binaries (the one you list after `install`) are made available in your PATH
* `pipx install --include-dependencies`: the main package AND every dependencies binaries are made available in your PATH.

When you install a package that has binaries that conflict with current binaries, then pipx will still install the package into its own directory, and will try to link as many of its binaries as possible into your PATH, but will not overwrite existing binaries in your PATH.

The bottom line is that pipx will try to do as much as it can of what you requested, and merely gives you warnings about what it cannot do to avoid over-writing.  We choose to keep pipx simple by not trying to do advanced things like refusing to install a package if its binaries conflict.  Doing something like that right in all cases would add much more complexity in the code (and possible failure points).

**Comment by alanwilter** [comment]
Thanks for your detailed explanation. After using a bit longer and installed other packages I can see the reason of your arguments.
